File: src/javakotlinexamples/java/Conditions.java
package javakotlinexamples.java;

public class Conditions {

    public static void main(String[] args) {
        int num = (int) (Math.random() * 4f);
        System.out.println(String.format("%d ist %s",
                num,
                isEven(num) ? "gerade" : "ungerade"));

        switch (num) {
            case 0:
                System.out.println("0");
                break;
            case 2:
                System.out.println("2");
                break;
            default:
                System.out.println("1 oder 3");
        }
    }

    public static boolean isEven(int num) {
        return (num % 2) != 1;
    }
}

File: src/javakotlinexamples/kotlin/Conditions.kt
package javakotlinexamples.kotlin

fun main(args: Array<String>) {
    val num = (Math.random() * 4).toInt()
    println("$num ist ${if (isEven(num)) "gerade" else "ungerade"}")

    when (num) {
        0 -> println("0")
        2 -> println("2")
        else -> println("1 oder 3")
    }

    val person = object {
        val name = null
    }
    val s = person.name ?: "???"
    println(s)
}

fun isEven(num: Int): Boolean {
    return (num % 2) != 1
}